contracts/wallet/BaseWallet.sol

```diff
@@ -101,16 +101,23 @@ contract BaseWallet {
      * @param _value The value of the transaction.
      * @param _data The data of the transaction.
      */
-    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly {
+    function invoke(address _target, uint _value, bytes calldata _data) external moduleOnly returns (bytes memory _result) {
+        bool success;
         // solium-disable-next-line security/no-call-value
-        (bool success, ) = _target.call.value(_value)(_data);
-        require(success, "BW: call to target failed");
+        (success, _result) = _target.call.value(_value)(_data);
+        if(!success) {
+            // solium-disable-next-line security/no-inline-assembly
+            assembly {
+                // _result = {result_length:32}{sig:4}{pos:32}{revert_reason_length:32}{revert_reason}
+                revert(add(_result, 0x64), add(_result, 0x44))
```

Comment:
Maybe we could use
```
assembly {
                returndatacopy(0, 0, returndatasize)
                revert(0, returndatasize)
            }
```
as we do in the fallback method as it is more readable.

Reply:
That would return the reason message prefixed with some junk because of the way reason messages are returned (see EIP 838). We could skip the junk using:
```
let len := sub(returndatasize, 0x44)
returndatacopy(0, 0x44, len)
revert(0, len)
```
I thought that this would be less gas-efficient than what I did (because of the additional copy) but it turns out that it's pretty much the same gas cost, so I can use that expression instead.